llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -2770,15 +2770,31 @@ bool llvm::replaceAllDbgUsesWith(Instruction &From, Value &To,
   return false;
 }
 
+void llvm::handleUnreachableTerminator(Instruction *I) {
+  // RemoveDIs: erase debug-info on this instruction manually.
+  I->dropDbgValues();
+  unsigned Opcode = I->getOpcode();
+  if (Opcode >= Instruction::Ret && Opcode <= Instruction::Invoke &&
+      I->getNumOperands() > 0) {
+    unsigned OpNum = isa<InvokeInst>(I) ? I->getNumOperands() - 1 : 0;
```

Comment:
You should loop over `I->operands()` here, not only update a single operand.

Reply:
you mean set the poison to all of operand?

Reply:
it has be done!